extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/connection/ConnectionService.java

```diff
@@ -251,8 +270,23 @@ private String getConnectionSearchFilter(String userDN,
             // The guacConfig group uses the seeAlso attribute to refer
             // to these other groups
             while (userRoleGroupResults.hasMore()) {
-                LDAPEntry entry = userRoleGroupResults.next();
-                connectionSearchFilter.append("(seeAlso=").append(escapingService.escapeLDAPSearchFilter(entry.getDN())).append(")");
+                try {
+                    LDAPEntry entry = userRoleGroupResults.next();
+                    connectionSearchFilter.append("(seeAlso=").append(escapingService.escapeLDAPSearchFilter(entry.getDN())).append(")");
+                }
+
+                catch (LDAPReferralException e) {
+                    if (confService.getFollowReferrals()) {
+                        logger.error("Could not follow referral.", e.getMessage());
+                        logger.debug("Error encountered trying to follow referral.", e);
+                        throw new GuacamoleServerException("Could not follow LDAP referral.", e);
```

Comment:
Recalling the changes proposed via #129, is it desirable that the entire search will abort with a failure if a referral error occurs? Or should the error be non-fatal and simply logged?

Reply:
My thinking is that if you explicitly enable referral following, an error following the referrals should cause an abort, and if you disable it, you can log it and ignore it.  I'm not opposed to a non-fatal logging, here, though, if that makes more sense.  I don't know off the top of my head what the generally accepted behavior is for other implementations.

In the case of the issue with Active Directory, disabling referral following completely allows you to query it via LDAP correctly - having it disabled doesn't cut off expected results or anything like that.

Reply:
@mike-jumper Any further thoughts on this, or opinions on which way it should go?  I'm not sure one way or the other - I do tend toward not wanting to break things if a single operation fails; however, I also think that if you're explicitly enabling it and it fails, you want a very clear indication that something is not working.  I see both sides, so I'm good going either way - it's easy enough to change down the road, and will get tweaked, anyway, when we switch LDAP APIs.